src/thing/get.ts

```diff
@@ -51,15 +56,15 @@ export function getUrl(
   property: Url | UrlString
 ): UrlString | null {
   internal_throwIfNotThing(thing);
-  const namedNodeMatcher = getNamedNodeMatcher(property);
-
-  const matchingQuad = findOne(thing, namedNodeMatcher);
-
-  if (matchingQuad === null) {
+  if (!internal_isValidUrl(property)) {
+    throw new ValidPropertyUrlExpectedError(property);
+  }
+  const predicateUrl = internal_toIriString(property);
+  const firstUrl = thing.predicates[predicateUrl]?.namedNodes?.[0] ?? null;
+  if (firstUrl === null) {
     return null;
   }
-
-  return matchingQuad.object.value;
+  return isLocalNodeIri(firstUrl) ? `#${getLocalNodeName(firstUrl)}` : firstUrl;
```

Comment:
\``#${getLocalNodeName(firstUrl)}`\` is going to be a construct that may be duplicated a bunch of times. It it worth, for readability, to factor it out in a function `serializeLocalNodeName` ?

Reply:
I think it only occurs here. It's a bit of an odd one out too, because most of the time local references should already be fully resolved after saving the SolidDataset. The only time this occurs is when you call `getIri` (or `getIriAll`) to obtain a URL you just added to a Thing - if you've fetched it, or already saved it, you'll get the full URL, and our own codebase just works with the skolemised URL.